src/gui/folderstatusmodel.cpp

```diff
@@ -698,17 +696,20 @@ void FolderStatusModel::slotLscolFinishedWithError(QNetworkReply* r)
         qDebug() << r->errorString();
         parentInfo->_lastErrorString = r->errorString();
 
+        parentInfo->resetSubs(this, idx);
+
         if (r->error() == QNetworkReply::ContentNotFoundError) {
             parentInfo->_fetched = true;
         } else {
-            if (!parentInfo->hasLabel()) {
+            bool add = !parentInfo->hasLabel();
+            if (add) {
```

Comment:
The problem was the lack of remove i think  (altough i tried to reproduce the crash localy but i could not)

So the problem would appear in the case there was a fetchLabel,  but that we get then a 404 so we would remove the label, but not call removeRows.  In this case, it is possible that the treeview still call us with an index that is child of this element, with row 0 (which was corresponding before to the label)  But now that the label is gone, the row 0 correspond to the first element of the _subs vector.  And since the vector is empty, this leads to crashes.


But you areright that we don't need to test for hasLabel